bundles/org.openhab.binding.ahawastecollection/src/main/java/org/openhab/binding/ahawastecollection/internal/AhaWasteCollectionHandler.java

```diff
@@ -0,0 +1,207 @@
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (command instanceof RefreshType) {
+            this.scheduler.execute(this::updateCollectionDates);
+        } else {
+            this.logger.warn("The AHA Abfuhrkalender is a read-only binding and can not handle commands"); //$NON-NLS-1$
+        }
+    }
+
+    @Override
+    public void initialize() {
+        this.config = this.getConfigAs(AhaWasteCollectionConfiguration.class);
+
+        final String gemeinde = this.config.commune;
+        final String street = this.config.street;
+        final String hausnr = this.config.houseNumber;
+        final String hausnraddon = this.config.houseNumberAddon;
+        final String ladeort = this.config.collectionPlace;
+
+        if (gemeinde.isBlank() || street.isBlank() || hausnr.isBlank() || ladeort.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Parameters are mandatory and must be configured");
+            return;
+        }
+
+        this.collectionSchedule = scheduleFactory.create(gemeinde, street, hausnr, hausnraddon, ladeort);
+
+        this.updateStatus(ThingStatus.UNKNOWN);
+
+        this.scheduler.execute(() -> {
+            final boolean online = this.updateCollectionDates();
+            if (online) {
+                this.restartJob();
+            }
+        });
+    }
+
+    /**
+     * Schedules an job that updates the collection dates at midnight.
+     */
+    private void restartJob() {
+        this.logger.debug("Restarting jobs for thing {}", this.getThing().getUID());
+        this.monitor.lock();
+        try {
+            this.stopJob();
+            if (this.getThing().getStatus() == ThingStatus.ONLINE) {
+                this.dailyJob = this.cronScheduler.schedule(this::updateCollectionDates, DAILY_MIDNIGHT);
+                this.logger.debug("Scheduled {} at midnight", this.dailyJob);
+                // Execute daily startup job immediately
+                updateCollectionDates();
+            }
+        } finally {
+            this.monitor.unlock();
+        }
+    }
+
+    /**
+     * Stops all jobs for this thing.
+     */
+    private void stopJob() {
+        this.monitor.lock();
+        try {
+            final ScheduledCompletableFuture<?> job = this.dailyJob;
+            if (job != null) {
+                job.cancel(true);
+            }
+            this.dailyJob = null;
+        } finally {
+            this.monitor.unlock();
+        }
+    }
+
+    private boolean updateCollectionDates() {
+        final Map<WasteType, CollectionDate> collectionDates = this.cache.getValue();
+        if (collectionDates == null || collectionDates.isEmpty()) {
+            return false;
+        }
+
+        this.logger.debug("Retrieved {} collection entries.", collectionDates.size());
+        this.updateChannels(collectionDates);
+
+        this.updateStatus(ThingStatus.ONLINE);
```

Comment:
You might want to set the Thing online only if the retrival of the remote data actually suceeded. This will set the Thing online, also if the retrival from the cache suceeded.